src/manifold/deferred.clj

```diff
@@ -48,6 +54,121 @@
   (successValue [default])
   (errorValue [default]))
 
+(declare then-apply then-apply-async
+         then-accept then-accept-async
+         then-run then-run-async
+
+         then-combine then-combine-async
+         then-accept-both then-accept-both-async
+         run-after-both run-after-both-async
+
+         apply-to-either apply-to-either-async
+         accept-either accept-either-async
+         run-after-either run-after-either-async
+
+         then-compose then-compose-async
+
+         then-handle then-handle-async
+         then-exceptionally
+
+         to-completable-future
+
+         when-complete when-complete-async)
```

Comment:
Is this necessary? Is there some mutual recursion I'm missing? Unless impossible, it's better to rearrange the code to avoid declarations.

Reply:
OK. But if we remove the need for SuccessDeferred, this should be revisited

Reply:
Looks like we still have a circular dep with ADeferred, ->deferred, and onto. I might still rearrange this later, but I'm resolving this for now.